src/java/org/apache/cassandra/db/tries/TrieSet.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.tries;
+
+import org.apache.cassandra.utils.bytecomparable.ByteComparable;
+
+/// A trie that defines an infinite set of `ByteComparable`s. The convention of this package is that sets always
+/// include all boundaries, all prefixes that lead to a boundary, and all descendants of all boundaries. This is done
+/// to properly define reverse iteration where prefixes are listed before their descendants, and to allow for the
+/// retrieval of metadata on paths pointing to specific keys.
+///
+/// Trie sets represent sets of ranges of coverage by listing the boundaries between them, and providing a way to
+/// identify the "covering state" for positions that are being skipped to.
+///
+/// Trie sets can be constructed from ranges using [#singleton], [#range] and [#ranges], and can be manipulated via
+/// the set algebra methods [#union], [#intersection] and [#weakNegation].
+public interface TrieSet extends CursorWalkable<TrieSetCursor>
+{
+    static TrieSet singleton(ByteComparable.Version version, ByteComparable b)
+    {
+        return dir -> new RangesCursor(dir, version, b, b);
+    }
+
+    static TrieSet range(ByteComparable.Version version, ByteComparable left, ByteComparable right)
```

Comment:
It seems to be the same as `ranges(version, left, right)`. Do we really need it ?